src/api/auth.ts

```diff
@@ -16,16 +16,56 @@ const postRequest = async (path: string, data: object) => {
   return response.json();
 };
 
-export const setToken = (walletAddress: string, token: string) =>
-  localStorage.setItem(`${TOKEN_STORAGE}-${walletAddress}`, token);
-export const getToken = (walletAddress: string) =>
-  localStorage.getItem(`${TOKEN_STORAGE}-${walletAddress}`);
-export const clearToken = (walletAddress: string) =>
+export const setToken = (walletAddress: string, token: string) => {
+  const storedTokens = localStorage.getItem(TOKEN_STORAGE);
+
+  if (storedTokens) {
+    const parsedStoredTokens = JSON.parse(storedTokens);
+
+    parsedStoredTokens[`${TOKEN_STORAGE}-${walletAddress}`] = token;
+
+    localStorage.setItem(TOKEN_STORAGE, JSON.stringify(parsedStoredTokens));
+  }
```

Comment:
What if we don't already have a seeded object in the local storage?

Please rewrite this to cover that case as well.